mycore-restapi/src/main/java/org/mycore/restapi/MCRRestAuthorizationFilter.java

```diff
@@ -93,14 +95,20 @@ private void checkRestAPIAccess(ContainerRequestContext requestContext, MCRRestA
     }
 
     private void checkBaseAccess(ContainerRequestContext requestContext, MCRRestAPIACLPermission permission,
-        String objectId, String derId, String path)
-        throws MCRRestAPIException {
-        LogManager.getLogger().debug("Permission: {}, Object: {}, Derivate: {}, Path: {}", permission, objectId, derId,
-            path);
-        Optional<String> checkable = Optional.ofNullable(derId)
-            .filter(d -> path != null) //only check for derId if path is given
-            .map(Optional::of)
-            .orElseGet(() -> Optional.ofNullable(objectId));
+        String objectId, String derId, String path) throws MCRRestAPIException {
+        Optional<String> optObjId = Optional.ofNullable(MCRRestAPIObjectsHelper.retrieveMCRObjectID(objectId))
+            .map(MCRObjectID::toString);
+        Optional<String> optDerId = Optional.empty();
+        if (optObjId.isPresent()) {
+            optDerId = Optional
+                .ofNullable(
+                    MCRRestAPIObjectsHelper.retrieveMCRDerivateID(MCRObjectID.getInstance(optObjId.get()), derId))
```

Comment:
Write a [PreMatching Filter](https://www.logicbig.com/tutorials/java-ee-tutorial/jax-rs/filters-pre-matching.html) and register it within `org.mycore.restapi.v1.MCRRestV1App` to rewrite the URL on-the-fly without tighter coupling (read: leave `MCRRestAuthorizationFilter` untouched)